### PR TITLE
ICU-22301 Add azure CI tests to be run post merge

### PR DESCRIPTION
**Pull request by rp9-next**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22301
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Comment by FrankYFTang** [comment]
How can we test .azure-post-merge-tests.yml in this PR.
Is there a way to invoke it by "/azp run", I try "/azp list" and there are no pipeline config for PostMerge

**Comment by rp9-next** [comment]
> How can we test .azure-post-merge-tests.yml in this PR. Is there a way to invoke it by "/azp run", I try "/azp list" and there are no pipeline config for PostMerge

I will have to create a pipeline in the Azure dashboard once these merged into main. 
I tested the ".yml" file using a separate pipline in my local fork ([PosMerge](https://dev.azure.com/ms/icu/_build?definitionId=605) & [PostMerge2](https://dev.azure.com/ms/icu/_build?definitionId=606&_a=summary)). 

I will create a similar pipeline after changes are merged. After that, this pipeline would trigger automatically whenever a code is committed to main/maint branches. Sadly, since we want them to run Post Merge, we have disabled validating PRs and triggering it via comments won't be possible.

File: .ci-builds/.azure-post-merge-tests.yml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    - main
+    - maint/*
+
+
+jobs:
+
+#-------------------------------------------------------------------------
+# VS 2022 Builds for testing C Samples on Windows
+#-------------------------------------------------------------------------
+- job: ICU4C_Samples_MSVC_x64_Debug
```
[the remark this answers is not in the document]
Comment (rp9-next) [reply .ci-builds/.azure-post-merge-tests.yml]: Thanks for the suggestion! Didn't knew we could do that :) Modified accordingly.

**Review by echeran — APPROVED** [review]
lgtm

**Comment by markusicu** [comment]
Hi @rp9-next I merged your PR before the code freeze. It looked ready. Sorry if you wanted to make more changes -- in that case please either roll back or fix-forward.

**Comment by FrankYFTang** [comment]
Can we document about how to run this somewhere?

**Pull request merged**